lib/type/list.ex

```diff
@@ -0,0 +1,156 @@
+defmodule Litmus.Type.List do
+  @moduledoc false
+
+  alias Litmus.Required
+  alias Litmus.Type
+
+  defstruct [
+    :min_length,
+    :max_length,
+    :length,
+    :type,
+    required: false
+  ]
+
+  @type t :: %__MODULE__{
+          min_length: non_neg_integer | nil,
+          max_length: non_neg_integer | nil,
+          length: non_neg_integer | nil,
+          type: String.t() | atom | nil,
+          required: boolean
+        }
+
+  @spec validate_field(t, String.t(), map) :: {:ok, map} | {:error, String.t()}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- check_if_list(type, field, data),
+         {:ok, data} <- type_validate(type, field, data),
+         {:ok, data} <- min_length_validate(type, field, data),
+         {:ok, data} <- max_length_validate(type, field, data),
+         {:ok, data} <- length_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec check_if_list(t, String.t(), map) :: {:ok, map} | {:error, String.t()}
+  defp check_if_list(%__MODULE__{}, field, params) do
```

Comment:
`validate_list`? or is there another name. typically you don't use `is` in Elixir function names